hawc/apps/riskofbias/serializers.py

```diff
@@ -125,18 +130,127 @@ class Meta:
+            data["study"] = study
+
+            # ...and same for author_id
+            author_id = tryParseInt(self.initial_data["author_id"], -1)
+            author = None
+            try:
+                author = HAWCUser.objects.get(id=author_id)
+            except ObjectDoesNotExist:
+                raise serializers.ValidationError("Invalid author_id")
+
+            data["author"] = author
+
+            assessment = study.assessment
+            # we also checked permissions for the caller, not just the supplied author, in api.py
+            if not study.user_can_edit_study(assessment, author):
+                # author is the user specified by the "author_id" value in the payload
+                raise serializers.ValidationError(
+                    "Author '%s' has invalid permissions to edit Risk of Bias for this study"
+                    % author
+                )
+
+            # Validation should check that:
+            # (1) no other active=True, final=True exists for this study_id, and
+            # (2) for scores, a default=True exists for each required metric (by study type) for a study.
+
+            # (1) does the study already have an active=True/final=True rob?
+            if self.initial_data["active"] is True and self.initial_data["final"] is True:
+                robs = study.riskofbiases.all()
+                for rob in robs:
+                    if rob.active and rob.final:
+                        raise serializers.ValidationError(
+                            "create failed; study %s already has an active & final risk of bias"
+                            % study_id
+                        )
+
+            # (2) subject to the settings of the metrics defined for this assessment,
+            # and the study_type of the study, we need to make sure all necessary scores
+            # have been submitted with a default=True
+            scores = self.initial_data["scores"]
+            required_metrics = models.RiskOfBiasMetric.objects.get_required_metrics(
+                assessment, study
+            )
+            problematic_scores = []
+            for metric in required_metrics:
+                domain = metric.domain
+                # there could be multiple scores for a given metric if there are overrides, so we need to fetch them all
+                scores_for_metric = [
+                    score
+                    for score in scores
+                    if "metric_id" in score and score["metric_id"] == metric.id
+                ]
+
+                metric_descriptor = "'%s:%s'" % (domain.name, metric.name)
+
+                if 0 == len(scores_for_metric):
+                    problematic_scores.append(
+                        "No score for metric %s/%s was submitted" % (metric.id, metric_descriptor)
+                    )
+                else:
+                    default_score_for_this_metric_included = False
+                    for submitted_score in scores_for_metric:
```

Comment:
only one score can be the default for a metric; can this check that the `len(default_scores)` equals exactly 1 for each metric else error?